syntax = "proto3";
package sandbox_grpc;

message RunCodeRequest {
    string code = 1;
}

message Result {
    string text = 1;
    optional string image = 2;
}

message RunCodeResponse {
    repeated Result results = 1;
    string stdout = 2;
    string stderr = 3;
}

message HealthcheckRequest {}

message HealthcheckResponse {}

service Sandbox {
    rpc RunCode(RunCodeRequest) returns (RunCodeResponse);
    rpc Healthcheck(HealthcheckRequest) returns (HealthcheckResponse);
}